src/sat/sat_clause.cpp

```diff
@@ -123,53 +123,80 @@ namespace sat {
 
     clause_allocator::clause_allocator():
         m_allocator("clause-allocator") {
-#ifdef _AMD64_
+#if defined(_AMD64_) 
         m_num_segments = 0;
 #endif
     }
 
     clause * clause_allocator::get_clause(clause_offset cls_off) const {
-#ifdef _AMD64_
-        return reinterpret_cast<clause *>(m_segments[cls_off & c_aligment_mask] + (static_cast<size_t>(cls_off) & ~c_aligment_mask));
+#if defined(_AMD64_) 
+#if defined (Z3DEBUG)
```

Comment:
This really deserves a comment explaining why we are doing this (i.e. mention ASan).